jujugui/static/gui/src/app/components/deployment-bar/deployment-bar.js

```diff
@@ -21,17 +21,59 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 YUI.add('deployment-bar', function() {
 
   juju.components.DeploymentBar = React.createClass({
+    previousNotifications: [],
 
-  /**
-    Get the label for the deploy button.
+    /**
+      Get the current state of the deployment bar.
 
-    @method _getDeployButtonLabel
-    @param {Boolean} hasCommits Does the env have commits.
-    @returns {String} the label for the deploy button
-  */
-  _getDeployButtonLabel: function(hasCommits) {
-    return hasCommits ? 'Commit changes' : 'Deploy changes';
-  },
+      @method getInitialState
+      @returns {Object} The current state.
+    */
+    getInitialState: function() {
+      return {
+        latestChangeDescription: null
+      };
+    },
+
+    componentDidMount: function() {
+      this._updateLatestChange(this.props.currentChangeSet);
+    },
+
+    componentWillReceiveProps: function(nextProps) {
```

Comment:
Take a look here: https://github.com/juju/juju-gui/blob/pyramid-fork/jujugui/static/gui/src/app/components/search-results/test-search-results.js#L342-L349

And here: https://github.com/juju/juju-gui/blob/pyramid-fork/jujugui/static/gui/src/app/components/search-results/test-search-results.js#L335-L340

I tested them by calling directly and mocking out anything they called, so much more of a true unit test.

That said, I'm not sure how @hatched feels about that approach ;-)